packages/components/src/client/components/ArtPlayer.ts

```diff
@@ -0,0 +1,211 @@
+/* eslint-disable vue/no-unused-properties */
+import { PropType, defineComponent, h, onBeforeUnmount, onMounted } from "vue";
+import { deepMerge } from "vuepress-shared/client";
+import { useSize } from "../composables/index.js";
+import { autoType, flv, m3u8, mpd } from "../utils/index.js";
+
+import type { VNode } from "vue";
+
+import type {
+  ArtPlayer,
+  ArtPlayerOptions,
+  ComponentsArtPlayerOptions,
+  CustomArtPlayerOptions,
+} from "../../shared/index.js";
+
+declare const ART_PLAYER: ComponentsArtPlayerOptions;
+
+const optionsDefault = {
+  width: "100%",
+  ratio: 16 / 9,
+  src: {
+    fullscreen: true,
+    autoSize: true,
+    setting: true,
+    whitelist: ["*"],
+  },
+};
+
+export default defineComponent({
+  name: "ArtPlayer",
+
+  props: {
+    /**
+     * artplayer src
+     */
+    src: {
+      type: Object as PropType<CustomArtPlayerOptions>,
+      default: optionsDefault.src,
+      required: false,
+    },
+
+    /**
+     * artplayer src.url
+     */
+    url: {
+      type: String,
+      default: "",
+      required: false,
+    },
+
+    /**
+     * artplayer src.poster
+     */
+    poster: {
+      type: String,
+      default: "",
+      required: false,
+    },
+
+    /**
+     * artplayer src.title
+     */
+    title: {
+      type: String,
+      default: "",
+      required: false,
+    },
+
+    /**
+     * artplayer src.muted
+     */
+    muted: {
+      type: Boolean,
+      default: false,
+      required: false,
+    },
+
+    /**
+     *  return artplayer
+     */
+    player: {
+      type: Function,
+      default: undefined,
+      required: false,
+    },
+
+    /**
+     * artplayer danmuku
+     */
+    pluginDanmuKu: {
+      type: Object,
```

Comment:
What's this, it should have a exact type with `Object as PropType<Type>`

Reply:
I do not think all artPlayer options could be supported, so you should create an actual type for options.

Reply:
> I do not think all artPlayer options could be supported, so you should create an actual type for options.

OK, add a `shims-artplayer.d.ts`